src/SystemPaths.cc

```diff
@@ -165,20 +203,43 @@ std::string SystemPaths::FindSharedLibrary(const std::string &_libName)
 void SystemPaths::SetFilePathEnv(const std::string &_env)
 {
   this->dataPtr->filePathEnv = _env;
+  std::string result;
+
   if (!this->dataPtr->filePathEnv.empty())
   {
     this->ClearFilePaths();
-    std::string result;
     if (env(this->dataPtr->filePathEnv, result))
     {
       this->AddFilePaths(result);
     }
   }
+  // TODO(CH3): Deprecated. Remove on tock.
+  else if (!this->dataPtr->filePathEnvDeprecated.empty())
```

Comment:
This doesn't look right, we've never assigned

```cpp
this->dataPtr->filePathEnvDeprecated = _env;
```

Reply:
Right, I'll emit a warning using string comparison

Reply:
https://github.com/gazebosim/gz-common/pull/365/commits/cb5f31e2958f927c0bcab1db07114fd24df82668